### DISCUSSION TITLE
Single touch not working anymore after surface kernel upgrade

### DISCUSSION THREAD

**Discussion by AnotherInfiniteLoop** [linux-surface/linux-surface#1011] [Q&A]
uname -a:
Linux XYZ 6.0.12-surface #5 SMP PREEMPT_DYNAMIC Sun Dec 11 14:19:44 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

About a month ago I installed Zorin OS (Ubuntu based) and linux-surface on my new Surface Pro 8. Got everything working, except multitouch. But didn't really miss that (yet). For single touch I had to install the ithc driver from: https://github.com/quo/ithc-linux.

After the surface kernel upgrade from 6.0.8 to the current version, single touch stopped working. Also Virtualbox stopped working and returns this error message when I start a VM: Kernel driver not installed (rc=-1908)
I tried all kinds of things and decided to start with a fresh Zorin OS install. But still the same outcome: no single touch (and still same error with VirtualBox)

I use secureboot and the strange thing is that during installation of linux-surface this never happened (tried a fresh install multiple times):
_"This package will print instructions to the terminal. In case you missed those: It will ask you to reboot. Then, upon reboot, a blue menu should pop up, asking you whether you want to enroll the key. Confirm with ok/yes and when asked for a password enter surface. In case you missed the menu or accidentally chose the wrong options, you can uninstall (fully) and then reinstall the package and reboot to trigger the menu again."_

Tried to reinstall the ithc driver from https://github.com/quo/ithc-linux, but no success.

Any idea how I can troubleshoot the single touch issue?

**Comment by NP-chaonay** [comment]
1) have you enbled "Lock Boot Configuration" in Surface UEFI?
2) try this
- 1) disable Secure Boot
- 2) resetup the Secure Boot (uninstall that  linux-surface-secureboot-mok first by using apt-purge) 
- 3) after finished setup in blue menu, then enable secureboot to 3rd party
- 4) setup secureboot for DKMS and then proceed install driver including ithc

3) if that all success then we will going to solve multitouch

**Reply by AnotherInfiniteLoop** [reply]
Thanks you or taking the time to help me.
I finished steps 1,2 and 3. I'm now trying to install the ithc driver from step 4. I get an error when executing this statement: 'sudo make dkms-install'. 

This is what I did to install ithc driver. Sorry for the auto markup of the editor, hope you can read it:

average@JOE:$ uname -a
Linux XYZ 6.0.12-surface #5 SMP PREEMPT_DYNAMIC Sun Dec 11 14:19:44 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
average@JOE:$ cd /Downloads/
average@JOE:/Downloads$ git clone https://github.com/quo/ithc-linux.git
Cloning into 'ithc-linux'...
remote: Enumerating objects: 93, done.
remote: Counting objects: 100% (93/93), done.
remote: Compressing objects: 100% (64/64), done.
remote: Total 93 (delta 58), reused 63 (delta 28), pack-reused 0
Unpacking objects: 100% (93/93), 28.45 KiB | 1.05 MiB/s, done.
average@JOE:/Downloads$ cd ithc-linux
average@JOE:/Downloads/ithc-linux$ sudo make dkms-install
test -e /usr/src/`sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`-`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf` && make dkms-uninstall
make[1]: Entering directory '/home/average/Downloads/ithc-linux'
modprobe -r ithc
modprobe: FATAL: Module ithc not found.
make[1]: [Makefile:30: dkms-uninstall] Error 1 (ignored)
dkms uninstall `sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`/`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`
Error! The module ithc 0.1 is not currently installed.
This module is not currently ACTIVE for kernel 6.0.12-surface (x86_64).
make[1]: [Makefile:31: dkms-uninstall] Error 5 (ignored)
dkms remove `sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`/`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`
Error! Invalid number of parameters passed.
Usage: remove <module>/<module-version> --all
   or: remove <module>/<module-version> -k <kernel-version>
make[1]: [Makefile:32: dkms-uninstall] Error 1 (ignored)
rm -rf /usr/src/`sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`-`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`
sync
make[1]: Leaving directory '/home/average/Downloads/ithc-linux'
mkdir -p /usr/src/`sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`-`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`
cp -r dkms.conf Makefile src /usr/src/`sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`-`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`
dkms add `sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`/`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`
Error! DKMS tree already contains: ithc-0.1
You cannot add the same module/version combo more than once.
make: *** [Makefile:24: dkms-install] Error 3

**Reply by NP-chaonay** [reply]
@AnotherInfiniteLoop try running `sudo dkms remove --all ithc/0.1` and then `sudo dkms status` pls

**Reply by AnotherInfiniteLoop** [reply]
@NP-chaonay Just removed ithc. 
After `sudo dkms status` I get this:
virtualbox, 6.1.38, 5.15.0-56-generic, x86_64: installed

**Reply by AnotherInfiniteLoop** [reply]
In the meantime tried this again: `sudo make dkms-install`:

test -e /usr/src/`sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`-`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf` && make dkms-uninstall
make[1]: Entering directory '/home/average/Downloads/ithc-linux'
modprobe -r ithc
modprobe: FATAL: Module ithc not found.
make[1]: [Makefile:30: dkms-uninstall] Error 1 (ignored)
dkms uninstall `sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`/`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`
Error! There are no instances of module: ithc
located in the DKMS tree.
make[1]: [Makefile:31: dkms-uninstall] Error 2 (ignored)
dkms remove `sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`/`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`
Error! Invalid number of parameters passed.
Usage: remove <module>/<module-version> --all
   or: remove <module>/<module-version> -k <kernel-version>
make[1]: [Makefile:32: dkms-uninstall] Error 1 (ignored)
rm -rf /usr/src/`sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`-`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`
sync
make[1]: Leaving directory '/home/average/Downloads/ithc-linux'
mkdir -p /usr/src/`sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`-`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`
cp -r dkms.conf Makefile src /usr/src/`sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`-`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`
dkms add `sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`/`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`

Creating symlink /var/lib/dkms/ithc/0.1/source ->
                 /usr/src/ithc-0.1

DKMS: add completed.
dkms build `sed -n '/^PACKAGE_NAME="\(.*\)"$/s//\1/p' dkms.conf`/`sed -n '/^PACKAGE_VERSION="\(.*\)"$/s//\1/p' dkms.conf`

Kernel preparation unnecessary for this kernel.  Skipping...

Building module:
cleaning build area...
make -j8 KERNELRELEASE=6.0.12-surface KDIR=/lib/modules/6.0.12-surface/build...(bad exit status: 2)
Error! Bad return status for module build on kernel: 6.0.12-surface (x86_64)
Consult /var/lib/dkms/ithc/0.1/build/make.log for more information.
make: *** [Makefile:25: dkms-install] Error 10





**Contents of this logfile: /var/lib/dkms/ithc/0.1/build/make.log. Check highlighted (bold) lines:**

DKMS make.log for ithc-0.1 for kernel 6.0.12-surface (x86_64)
di 20 dec 2022 10:57:30 CET
make[1]: Entering directory '/var/lib/dkms/ithc/0.1/build'
mkdir -p build
find build/ -type l -exec rm {} +
ln -sr src/* build/
make -C /lib/modules/6.0.12-surface/build M=/var/lib/dkms/ithc/0.1/build/build
make[2]: Entering directory '/usr/src/linux-headers-6.0.12-surface'
**warning: the compiler differs from the one used to build the kernel
  The kernel was built by: gcc (Ubuntu 11.3.0-1ubuntu1-22.04) 11.3.0
  You are using:           gcc (Ubuntu 9.4.0-1ubuntu1-20.04.1) 9.4.0**
  CC [M]  /var/lib/dkms/ithc/0.1/build/build/ithc-main.o
  CC [M]  /var/lib/dkms/ithc/0.1/build/build/ithc-regs.o
  CC [M]  /var/lib/dkms/ithc/0.1/build/build/ithc-dma.o
  CC [M]  /var/lib/dkms/ithc/0.1/build/build/ithc-api.o
  CC [M]  /var/lib/dkms/ithc/0.1/build/build/ithc-debug.o
gcc: error: unrecognized command line option ‘-mharden-sls=all’
gcc: error: unrecognized command line option ‘-mharden-sls=all’
gcc: error: unrecognized command line option ‘-mharden-sls=all’
gcc: error: unrecognized command line option ‘-mharden-sls=all’
gcc: error: unrecognized command line option ‘-mharden-sls=all’
gcc: error: unrecognized command line option ‘-fzero-call-used-regs=used-gpr’
make[3]: *** [scripts/Makefile.build:249: /var/lib/dkms/ithc/0.1/build/build/ithc-regs.o] Error 1
make[3]: *** Waiting for unfinished jobs....
gcc: error: unrecognized command line option ‘-fzero-call-used-regs=used-gpr’
make[3]: *** [scripts/Makefile.build:249: /var/lib/dkms/ithc/0.1/build/build/ithc-debug.o] Error 1
gcc: error: unrecognized command line option ‘-fzero-call-used-regs=used-gpr’
make[3]: *** [scripts/Makefile.build:249: /var/lib/dkms/ithc/0.1/build/build/ithc-main.o] Error 1
gcc: error: unrecognized command line option ‘-fzero-call-used-regs=used-gpr’
make[3]: *** [scripts/Makefile.build:249: /var/lib/dkms/ithc/0.1/build/build/ithc-dma.o] Error 1
gcc: error: unrecognized command line option ‘-fzero-call-used-regs=used-gpr’
make[3]: *** [scripts/Makefile.build:249: /var/lib/dkms/ithc/0.1/build/build/ithc-api.o] Error 1
make[2]: *** [Makefile:1858: /var/lib/dkms/ithc/0.1/build/build] Error 2
make[2]: Leaving directory '/usr/src/linux-headers-6.0.12-surface'
make[1]: *** [Makefile:8: default] Error 2
make[1]: Leaving directory '/var/lib/dkms/ithc/0.1/build'

**Reply by AnotherInfiniteLoop** [reply]
> @NP-chaonay Just removed ithc. After `sudo dkms status` I get this: virtualbox, 6.1.38, 5.15.0-56-generic, x86_64: installed

Hi @NP-chaonay, I executed the steps you suggested. What could be the next step?

**Reply by NP-chaonay** [reply]
@AnotherInfiniteLoop 
sorry for late response, according to lastest error:

your gcc that installed in OS is not supported for building this

the way is to try upgrade gcc or OS, or setup gcc manually (download and put in path of PATH env)
